src/StochasticNoiseSimulator.cpp

```diff
@@ -248,19 +256,47 @@ void StochasticNoiseSimulator::runStochSimulationForId(unsigned int
                     targets  = classic_op->getOperation()->getTargets();
                     controls = classic_op->getOperation()->getControls();
                     if (!execute_op) {
-                        // applyNoiseOperation(targets.front(), controls, identity_DD, (std::unique_ptr<dd::Package> &) localDD, localRootEdge, generator, dist, line, identity_DD);
+                        // applyNoiseOperation(targets.front(), controls, identityDD, (std::unique_ptr<dd::Package> &) localDD, localRootEdge, generator, dist, line, identityDD);
                         continue;
                     }
                 } else {
-                    dd_op    = dd::getDD(op.get(), localDD);
                     targets  = op->getTargets();
                     controls = op->getControls();
+
+                    if (targets.size() == 1 && controls.empty()) {
+                        dd_op = localDD->stochasticNoiseOperationCache.lookup(op->getType(), targets.front());
```

Comment:
camelCase. you might want to simply call such occurrences `op` or `operation`

Reply:
done